internal/service/rpz/record_rpz_cname_ipaddress_resource.go

```diff
@@ -0,0 +1,334 @@
+package rpz
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+
+	niosclient "github.com/infobloxopen/infoblox-nios-go-client/client"
+
+	"github.com/infobloxopen/terraform-provider-nios/internal/utils"
+)
+
+var readableAttributesForRecordRpzCnameIpaddress = "canonical,comment,disable,extattrs,is_ipv4,name,rp_zone,ttl,use_ttl,view,zone"
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ resource.Resource = &RecordRpzCnameIpaddressResource{}
+var _ resource.ResourceWithImportState = &RecordRpzCnameIpaddressResource{}
+
+func NewRecordRpzCnameIpaddressResource() resource.Resource {
+	return &RecordRpzCnameIpaddressResource{}
+}
+
+// RecordRpzCnameIpaddressResource defines the resource implementation.
+type RecordRpzCnameIpaddressResource struct {
+	client *niosclient.APIClient
+}
+
+func (r *RecordRpzCnameIpaddressResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_" + "rpz_record_rpz_cname_ipaddress"
```

Comment:
```suggestion
	resp.TypeName = req.ProviderTypeName + "_" + "rpz_record_cname_ipaddress"
```
Change required throughout examples and tests